knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -138,6 +145,25 @@ func Delete(instance *servingv1alpha1.KnativeServing, apiclient client.Client, s
 	return nil
 }
 
+// deleteDeprecatedResources removes deprecated resources created by previous versions
+func deleteDeprecatedResources(instance *servingv1alpha1.KnativeServing, apiclient client.Client) error {
+	metaName := metav1.ObjectMeta{
+		Name:      deprecatedResourceName,
+		Namespace: instance.Namespace,
+	}
+	toDelete := []runtime.Object{
+		&appsv1.Deployment{ObjectMeta: metaName},
+		&corev1.Service{ObjectMeta:	metaName},
```

Comment:
For posterity from discussion in Slack: This will probably remove the K8s Service created by the Knative Service to result in an error loop.

@dsimansk and I agreed to rename the new Knative Service to bypass this issue.

Reply:
I pushed an update with new name `kn-cli`. Meanwhile I've tested upgrade scenario `1.9->1.10` that passed successfully with old resources deleted, new created and URL in `ConsoleCLIDownload` CR updated.